sw/device/tests/crt_test.c

```diff
@@ -54,6 +55,116 @@ static void init_uart(void) {
   base_uart_stdout(&uart0);
 }
 
+/**
+ * Test that crt_section_clear correctly zeros word aligned sections.
+ *
+ * Sections are simulated using word aligned regions of various sizes within an
+ * array.
+ *
+ * Does not return if the test fails.
+ */
+static void test_crt_section_clear(void) {
+  // Function to test (symbol in the CRT assembly library).
+  extern void crt_section_clear(void *start, void *end);
+
+  // Maximum end index of target section.
+  const size_t kLen = 32;
+
+  // Section indices (start inclusive, end exclusive).
+  const struct {
+    size_t start;
+    size_t end;
+  } kTests[6] = {
```

Comment:
Nit: this array length can be left out. Up to you.

Reply:
Done